core/account-id/src/errors.rs

```diff
@@ -22,22 +24,32 @@ impl fmt::Display for ParseAccountError {
 }
 
 /// A list of errors that occur when parsing an invalid Account ID.
-#[derive(Eq, Hash, Clone, Debug, PartialEq)]
+#[non_exhaustive]
+#[derive(Eq, Clone, Debug, PartialEq)]
 pub enum ParseErrorKind {
     TooLong,
     TooShort,
     Invalid,
 }
 
 impl ParseErrorKind {
+    /// Returns `true` if the Account ID was too long.
     pub fn is_too_long(&self) -> bool {
         matches!(self, ParseErrorKind::TooLong)
     }
 
+    /// Returns `true` if the Account ID was too short.
     pub fn is_too_short(&self) -> bool {
         matches!(self, ParseErrorKind::TooShort)
     }
```

Comment:
I don't think we need these two methods, the caller can `matches!` or `==` just fine.

Reply:
Yeah, initially I thought it could save the extra `ParseErrorKind` import if you could check `err.kind().is_too_long()`. But going by how rarely it's going to be used, it does make sense to drop it.